data/transactions/logic/eval.go

```diff
@@ -2119,6 +2159,49 @@ func opSwitch(cx *EvalContext) error {
 	return nil
 }
 
+func opMatch(cx *EvalContext) error {
+	n := int(cx.program[cx.pc+1])
+	// stack contains the n sized match list and the single match value
+	if n+1 > len(cx.stack) {
+		return fmt.Errorf("match expects %d stack args while stack only contains %d", n+1, len(cx.stack))
+	}
+
+	last := len(cx.stack) - 1
+	matchVal := cx.stack[last]
+	cx.stack = cx.stack[:last]
+
+	argBase := len(cx.stack) - n
+	matchList := cx.stack[argBase:]
+	cx.stack = cx.stack[:argBase]
+
+	// first we must verify that the matchList has the expected length and all types match the type of matchVal
+	if len(matchList) != n {
+		return fmt.Errorf("list of match constants has length %d when length %d was expected", len(matchList), n)
+	}
```

Comment:
This is impossible, right?  You've confirmed the stack is big enough, then sliced it up properly.

(Certainly the comment is out of date.)

Reply:
You're right, the check and comment are both out of date.